### PR TITLE
add: backoff reconnect to sink

### PR DESCRIPTION
**Pull request by digikata**
issue #116

### CODE REVIEW

**Comment by digikata** [comment]
Manually tested w/ `make integration_tests`.  Local tests pass

File: crates/sql-sink/src/backoff.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+use std::time::Duration;
+
+pub(crate) enum Backoff {
+    Fib(u64, u64),
```
Comment (galibey) [inline crates/sql-sink/src/backoff.rs:4]: We already have such thing in the common library https://docs.rs/fluvio-future/latest/fluvio_future/retry/struct.FibonacciBackoff.html

File: crates/sql-sink/src/main.rs [hunk 2]
```diff
@@ -1,24 +1,58 @@
...
-        trace!(?operation);
-        sink.send(operation).await?;
+    let mut backoff = Backoff::new();
+
+    loop {
+        let wait = backoff.next();
+
+        if wait > BACKOFF_LIMIT {
+            error!("Max retry reached, exiting");
+        }
```
Comment (galibey) [inline crates/sql-sink/src/main.rs:38]: Judging by the error message, there is a missing `return` here.

File: crates/sql-sink/src/main.rs [hunk 3]
```diff
@@ -1,24 +1,58 @@
...
+        let sink = SqlSink::new(&config)?;
+        let mut sink = match sink.connect(None).await {
+            Ok(sink) => sink,
+            Err(err) => {
+                warn!(
+                    "Error connecting to streaming source: \"{}\", reconnecting in {}.",
```
Comment (galibey) [inline crates/sql-sink/src/main.rs:44]: Is the "streaming source" the right phrase here? It's more of "sink" or "destination".

**Review by galibey — CHANGES_REQUESTED** [review]

File: crates/sql-sink/src/main.rs (hunk 2 above)
Comment (digikata) [reply crates/sql-sink/src/main.rs:43]: Adjusted the backoff and error message so the connector will continue retrying - though we may need a long term company discussion on where we we want to default.

 I also added a reset of t he backoff time to a short timeframe if a successful connect occurs

**Review by galibey — APPROVED** [review]
LGTM

**Pull request merged**